app/assets/javascripts/browsered/web-payments/google-pay.js

```diff
@@ -1,37 +1,55 @@
 'use strict'
 
-const { prepareRequestObject } = require('./helpers')
-
-module.exports = () => {
-  const { supportedInstruments, details, options } = prepareRequestObject('standard')
-  const request = new PaymentRequest(supportedInstruments, details, options)
-
-  request.show()
-    .then(result => {
-      const payload = result.toJSON()
-      payload.csrfToken = document.getElementById('csrf').value
-      payload.chargeId = window.paymentDetails.chargeID
-      return fetch(`/payment-request/${window.paymentDetails.chargeID}`, {
-        method: 'POST',
-        redirect: 'follow',
-        credentials: 'same-origin',
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify(payload)
-      }).then(response => {
-        if (response.status === 200) {
-          result.complete('success')
-          return window.location.href = response.url
-        } else {
-          result.complete('fail')
-          return window.location.href = response.url
-        }
-      }).catch(() => {
-        result.complete('fail')
-        return window.location.href = response.url
-      })
-    }).catch(err => {
-      return err
+const { getGooglePaymentsConfiguration, showErrorSummary } = require('./helpers')
+const { email_collection_mode } = window.Charge
+
+
+const processPayment = paymentData => {
+  // show returned data in developer console for debugging
+  console.log(paymentData);
+  // @todo pass payment token to your gateway to process payment
+  const paymentToken = paymentData.details.paymentMethodData.tokenizationData.token;
+  console.log('paymentToken', paymentToken);
+}
+
+const createGooglePaymentRequest = () => {
+  const methodData = [{
+    supportedMethods: 'https://google.com/pay',
+    data: getGooglePaymentsConfiguration()
+  }];
+
+  const details = {
+    total: {
+      label: window.paymentDetails.description,
+      amount: {
+        currency: 'GBP',
+        value: window.paymentDetails.amount
+      }
+    }
+  };
+
+  const options = {
+    requestPayerEmail: email_collection_mode !== 'OFF',
+    requestPayerName: true
+  };
+
+  return new PaymentRequest(methodData, details, options);
```

Comment:
does this PaymentRequest make the call to to google pay API?

Reply:
Yes this launches the native dialog and because here — 

https://github.com/alphagov/pay-frontend/pull/622/files#diff-2b4385fbba492188502b49e1137f1efbR17 

Google Pay is declared as a payment method, Chrome knows how to contact Google Pay